plugins/inputs/logfile/fileconfig.go

```diff
@@ -203,3 +251,55 @@ func (config *FileConfig) isMultilineStart(logValue string) bool {
 	}
 	return config.MultiLineStartPatternP.MatchString(logValue)
 }
+
+func (config *FileConfig) shouldFilterLog(event logs.LogEvent) bool {
```

Comment:
Funnily enough, your suggestion about negative naming was something I was mulling over myself, so I appreciate getting confirmation that it's something I should pull the trigger on. 

As far as emitting the matched lines, I think I'm doing something _close_ already for matching the number of dropped logs, and I just put it into the profiler. When you flip the agent to debug mode, you can see the profiler dump the dropped log count per log file.

Reply:
WRT the debug lines, I think a concern with using this functionality would be troubleshooting and supporting it. What are the consequences of being slow with processing the filters? How do we determine the filter processes are bottlenecking log publication? 

It was not entirely useful in my development because I wrote the benchmark unit tests which illustrate the information to me. I'm not opposed to dropping the elapsed time debug line. As far as the count metric for log messages, I am partial to tracking dropped log messages because technically, people who use CloudWatch today can already see the number of log events that get submitted to a log group.